lib/logging.dart

```diff
@@ -22,7 +22,7 @@ Level recordStackTraceAtLevel = Level.OFF;
 ///
 /// This will be the level of all loggers if [hierarchicalLoggingEnabled] is
 /// false.
-Level _rootLevel = Level.INFO;
+Level? _rootLevel = Level.INFO;
```

Comment:
It took me a bit of digging to figure out why this was made nullable.  Here's the reason, with details from the nullability graph:
- The type `Level` was associated with the nullability node `type(785)`.
- This nullability node was made nullable due to an edge from `type(3578)` to `type(785)`, caused by the assignment `_rootLevel = value` at line 114
- The node `type(3578)` was associated with the type of the setter parameter `value`.
- That nullability node, in turn, was made nullable due to an edge from `always` to `type(3578)`, caused by the assignment `a.level = null` at line 242 of `logging_test.dart`.

I think this illustrates the need for some further diagnostic output from the migration tool, or some sort of interactive way of running it.  It would be nice to be able to point at a `?` the tool added, and ask "where is the assignment that caused this to be made nullable?"  That information is present in the migration engine's nullability graph, but it's not currently exposed through the UI.

Reply:
What happens if we add a `/*!*/`?

Reply:
@natebosch 
> What happens if we add a `/*!*/`?

Three differences:
- No `?` is added here
- On line 98, the unnecessary null check is removed, changing it to `/* if (_level != null) */ return _level;`
- On line 107, a null check is added, changing it to `_level = value!;`

Reply:
Scratch that, I added the `/*!*/` in the wrong place.  What happens instead is:
- No `?` is added here
- On line 114, a null check is added, changing it to `_rootLevel = value!;`

Reply:
My hope would be that it chooses something different for line 105

`set level(Level? value) {`

I would prefer

`set level(Level value) {`

did the tool make this choice to try to keep the setter with the same API as before?